base/src/main/java/com/austindroids/austinfeedsme/data/eventbrite/EventbriteDataSource.kt

```diff
@@ -38,64 +36,45 @@ class EventbriteDataSource(val eventsRepository: EventsRepository) : EventsDataS
+
+        val eventbriteEventsSubscriber = Consumer<List<EventbriteEvent>> { eventbriteEvents ->
             val convertedEventbriteEvents = ArrayList<Event>()
-            for (eventbriteEvent in eventbriteEvents.getEvents()) {
+            for (eventbriteEvent in eventbriteEvents) {
                 convertedEventbriteEvents.add(EventbriteUtils.transformEventBrite(eventbriteEvent))
             }
-
-            cleanAndLoadEventbriteEvents(convertedEventbriteEvents, object : CleanCallback {
-                override fun loadCleanEvents(events: List<Event>) {
-                    callback.onEventsLoaded(events)
-                }
-            })
+            callback.onEventsLoaded(convertedEventbriteEvents)
         }
 
-        Single.concat(observableList)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(eventbriteEventsSubscriber, Consumer<Throwable> { t -> Timber.e(t) })
-    }
+        val listMerger = Function<Array<Any>, List<EventbriteEvent>> { eventBriteEvents ->
 
-    override fun saveEvent(eventToSave: Event, callback: EventsDataSource.SaveEventCallback) {
-
-    }
+            val mergedEvents: ArrayList<EventbriteEvent> = ArrayList()
 
-    private fun cleanAndLoadEventbriteEvents(events: MutableList<Event>, callback: CleanCallback) {
-        val callbackTimestamp = Date().time
-        Timber.d("onResponse: Event's from eventbrite $callbackTimestamp:${events.size}")
-
-        val eventbriteEventMap = HashMap<String, Event>()
-        for (event in events) {
-            if (event.time > callbackTimestamp) {
-                eventbriteEventMap[event.id] = event
+            eventBriteEvents.map { it as EventbriteEvents }.forEach { eventbriteEvents ->
+                eventbriteEvents.events.forEach {
+                    if (!eventIds.contains(it.id)) {
+                        eventIds.add(it.id)
+                        mergedEvents.add(it)
+                    }
+                }
             }
-        }
 
-        eventsRepository.getEvents(object: EventsDataSource.LoadEventsCallback {
-
-            override fun onEventsLoaded(events: MutableList<Event>?) {
-                events?.forEach {
-                    eventbriteEventMap.remove(it.id)
-                }
+            return@Function mergedEvents
 
-                callback.loadCleanEvents(ArrayList(eventbriteEventMap.values))
-            }
+        }
 
-            override fun onError(error: String?) {
-                Timber.e(error)
-                callback.loadCleanEvents(ArrayList())
-            }
-        }, false)
+        Single.zip(observableList, listMerger)
+                .subscribe(eventbriteEventsSubscriber, Consumer<Throwable> { Timber.e(it) })
     }
 
-    internal interface CleanCallback {
-        fun loadCleanEvents(events: List<Event>)
+    override fun saveEvent(eventToSave: Event, callback: EventsDataSource.SaveEventCallback) {
+
```

Comment:
Are you sure this is correct?

Reply:
will check it out

Reply:
yes - we don't have write access to eventbrite - probably need to remove the no-op implementations and create an interface that just includes get  and one that includes saving